File: src/系统/03．技能系统/05．单位技能/04．英雄技能/21．朱雀院红叶/07．D技能.lua
local ____lualib = require("lualib_bundle")
local __TS__Delete = ____lualib.__TS__Delete
local ____exports = {}
local _____79FB_9664D_72B6_6001, jass, removeDelayedCallback, _____79FB_9664_5355_4F4D_6307_5B9ABuff, _____9500_6BC1_5355_4F4D_5750_6807_8DDF_968F_7279_6548, debugLogForce, _____79D8_4F20BuffID, _____5200_73AF_7279_6548_952E, ____D_72B6_6001_8868
local ____00_FF0E_914D_7F6E = require("系统.03．技能系统.05．单位技能.04．英雄技能.21．朱雀院红叶.00．配置")
local _____6731_96C0_9662_7EA2_53F6_6280_80FD_914D_7F6E = ____00_FF0E_914D_7F6E["朱雀院红叶技能配置"]
local _____6731_96C0_9662_7EA2_53F6_8868_73B0_914D_7F6E = ____00_FF0E_914D_7F6E["朱雀院红叶表现配置"]
local _____6731_96C0_9662_7EA2_53F6_97F3_6548_914D_7F6E = ____00_FF0E_914D_7F6E["朱雀院红叶音效配置"]
local _____6731_96C0_9662_7EA2_53F6Buff_914D_7F6E = ____00_FF0E_914D_7F6E["朱雀院红叶Buff配置"]
local _____6731_96C0_9662_7EA2_53F6_52A8_4F5C_69FD = ____00_FF0E_914D_7F6E["朱雀院红叶动作槽"]
local _____6731_96C0_9662_7EA2_53F6_5F85_5E73_8861_6570_503C = ____00_FF0E_914D_7F6E["朱雀院红叶待平衡数值"]
function _____79FB_9664D_72B6_6001(_____82F1_96C4)
    if _____82F1_96C4 == nil or _____82F1_96C4 == 0 then
        return
    end
    local id = jass.GetHandleId(_____82F1_96C4)
    local _____72B6_6001 = ____D_72B6_6001_8868[id]
    if _____72B6_6001 == nil then
        return
    end
    if _____72B6_6001["到期回调ID"] ~= 0 then
        removeDelayedCallback(_____72B6_6001["到期回调ID"])
        _____72B6_6001["到期回调ID"] = 0
    end
    debugLogForce(
        "红叶-D",
        "Buff",
        "操作",
        "移除",
        "目标",
        _____82F1_96C4
    )
    _____9500_6BC1_5355_4F4D_5750_6807_8DDF_968F_7279_6548(_____82F1_96C4, _____5200_73AF_7279_6548_952E)
    _____79FB_9664_5355_4F4D_6307_5B9ABuff(_____82F1_96C4, _____79D8_4F20BuffID)
    __TS__Delete(____D_72B6_6001_8868, id)
end
jass = require("jass.common")
local ____require_result_0 = require("lib.扩展函数.封装函数.01．通用工具.01．FourCC转换安全版")
local stringToFourCCSafe = ____require_result_0.stringToFourCCSafe
local ____require_result_1 = require("系统.00．核心系统.05．中心计时器")
local getGameTime = ____require_result_1.getGameTime
local addDelayedCallback = ____require_result_1.addDelayedCallback
removeDelayedCallback = ____require_result_1.removeDelayedCallback
local ____require_result_2 = require("系统.03．技能系统.00．技能模板+函数.04．机制组件.10．复杂战斗通用机制.16．单位技能壳监听注册器")
local _____6CE8_518C_5355_4F4D_6280_80FD_58F3_76D1_542C = ____require_result_2["注册单位技能壳监听"]
local ____require_result_3 = require("系统.05．Buff系统.00．Buff系统")
local registerManualBuff = ____require_result_3.registerManualBuff
_____79FB_9664_5355_4F4D_6307_5B9ABuff = ____require_result_3["移除单位指定Buff"]
local ____require_result_4 = require("lib.扩展函数.封装函数.01．通用工具.03．特效")
local _____521B_5EFA_5355_4F4D_5750_6807_8DDF_968F_7279_6548 = ____require_result_4["创建单位坐标跟随特效"]
_____9500_6BC1_5355_4F4D_5750_6807_8DDF_968F_7279_6548 = ____require_result_4["销毁单位坐标跟随特效"]
local ____require_result_5 = require("lib.扩展函数.封装函数.02．音效系统.03．3D音效播放")
local Sound3DII_UnitPlayReuse = ____require_result_5.Sound3DII_UnitPlayReuse
local ____require_result_6 = require("系统.09．表现系统.10．英雄语音.10．技能喊话.01．英雄技能喊话")
local _____64AD_653E_82F1_96C4_6280_80FD_558A_8BDD = ____require_result_6["播放英雄技能喊话"]
local ____require_result_7 = require("系统.03．技能系统.05．单位技能.04．英雄技能.21．朱雀院红叶.02．被动效果")
local _____662F_6731_96C0_9662_7EA2_53F6 = ____require_result_7["是朱雀院红叶"]
local _____767B_8BB0_6731_96C0_9662_6E05_7406 = ____require_result_7["登记朱雀院清理"]
local _____6CE8_518C_7834_7EFD_65A9_76D1_542C = ____require_result_7["注册破绽斩监听"]
local _____64AD_653E_7EA2_53F6_52A8_4F5C = ____require_result_7["播放红叶动作"]
local ____require_result_8 = require("lib.扩展函数.自定义扩展函数.03．调试输出")
debugLogForce = ____require_result_8.debugLogForce
local _____82F1_96C4_5355_4F4D_7C7B_578BID = stringToFourCCSafe(_____6731_96C0_9662_7EA2_53F6_6280_80FD_914D_7F6E["单位类型ID"])
_____79D8_4F20BuffID = _____6731_96C0_9662_7EA2_53F6Buff_914D_7F6E["秘传三式"]
local ____D_914D_7F6E = _____6731_96C0_9662_7EA2_53F6_5F85_5E73_8861_6570_503C.D
local ____D_79D8_4F20_4E09_5F0F_97F3_6548 = _____6731_96C0_9662_7EA2_53F6_97F3_6548_914D_7F6E["D秘传三式"]
_____5200_73AF_7279_6548_952E = "朱雀院红叶D刀环"
____D_72B6_6001_8868 = {}
local function _____5237_65B0D_663E_793A(_____82F1_96C4, _____72B6_6001)
    local _____5269_4F59_79D2 = _____72B6_6001["到期时间"] - getGameTime()
    if _____5269_4F59_79D2 <= 0 then
        _____79FB_9664D_72B6_6001(_____82F1_96C4)
        return
    end
    registerManualBuff(
        _____82F1_96C4,
        _____79D8_4F20BuffID,
        _____5269_4F59_79D2,
        _____72B6_6001["剩余次数"],
        {stack = _____72B6_6001["剩余次数"]}
    )
end
local function _____5F00_542FD_79D8_4F20_4E09_5F0F(_context, _____65BD_6CD5_8005, ______6280_80FD_5B9E_4F8BID)
    debugLogForce("红叶-D", "释放", "技能实例ID", "-")
    if not _____662F_6731_96C0_9662_7EA2_53F6(_____65BD_6CD5_8005) then
        return
    end
    _____64AD_653E_7EA2_53F6_52A8_4F5C(_____65BD_6CD5_8005, _____6731_96C0_9662_7EA2_53F6_52A8_4F5C_69FD["D启动"])
    _____64AD_653E_82F1_96C4_6280_80FD_558A_8BDD(_____65BD_6CD5_8005, "朱雀院红叶", _____6731_96C0_9662_7EA2_53F6_6280_80FD_914D_7F6E.D["技能ID"])
    local id = jass.GetHandleId(_____65BD_6CD5_8005)
    local _____5DF2_6709 = ____D_72B6_6001_8868[id]
    if _____5DF2_6709 ~= nil then
        if _____5DF2_6709["到期回调ID"] ~= 0 then
            removeDelayedCallback(_____5DF2_6709["到期回调ID"])
        end
        _____5DF2_6709["到期时间"] = getGameTime() + ____D_914D_7F6E["持续秒"]
        _____5DF2_6709["到期回调ID"] = addDelayedCallback(
            ____D_914D_7F6E["持续秒"] * 1000,
            function()
                _____79FB_9664D_72B6_6001(_____65BD_6CD5_8005)
            end
        )
        _____5237_65B0D_663E_793A(_____65BD_6CD5_8005, _____5DF2_6709)
        return
    end
    local _____72B6_6001 = {
        ["剩余次数"] = ____D_914D_7F6E["强化次数"],
        ["到期时间"] = getGameTime() + ____D_914D_7F6E["持续秒"],
        ["延长次数"] = 0,
        ["到期回调ID"] = 0
    }
    _____72B6_6001["到期回调ID"] = addDelayedCallback(
        ____D_914D_7F6E["持续秒"] * 1000,
        function()
            _____79FB_9664D_72B6_6001(_____65BD_6CD5_8005)
        end
    )
    ____D_72B6_6001_8868[id] = _____72B6_6001
    debugLogForce("红叶-D", "状态", "开启秘传", _____72B6_6001["剩余次数"])
    _____5237_65B0D_663E_793A(_____65BD_6CD5_8005, _____72B6_6001)
    if _____6731_96C0_9662_7EA2_53F6_8868_73B0_914D_7F6E["D刀环"]["模型路径"] ~= "" then
        _____521B_5EFA_5355_4F4D_5750_6807_8DDF_968F_7279_6548(
            _____65BD_6CD5_8005,
            _____6731_96C0_9662_7EA2_53F6_8868_73B0_914D_7F6E["D刀环"]["模型路径"],
            _____5200_73AF_7279_6548_952E,
            _____6731_96C0_9662_7EA2_53F6_8868_73B0_914D_7F6E["D刀环"]["缩放"],
            _____6731_96C0_9662_7EA2_53F6_8868_73B0_914D_7F6E["D刀环"]["高度"],
            1,
            nil,
            0,
            _____6731_96C0_9662_7EA2_53F6_8868_73B0_914D_7F6E["D刀环"].RGB
        )
    end
    Sound3DII_UnitPlayReuse(____D_79D8_4F20_4E09_5F0F_97F3_6548["路径"], _____65BD_6CD5_8005, ____D_79D8_4F20_4E09_5F0F_97F3_6548["裁断距离"])
    _____767B_8BB0_6731_96C0_9662_6E05_7406(
        _____65BD_6CD5_8005,
        "红叶D",
        function()
            _____79FB_9664D_72B6_6001(_____65BD_6CD5_8005)
        end
    )
end
--- 尝试消费 1 次 D 强化（无 D 状态或次数不足返回 false，技能仍执行基础效果）
____exports["尝试消费D强化"] = function(_____82F1_96C4)
    if _____82F1_96C4 == nil or _____82F1_96C4 == 0 then
        return false
    end
    local _____72B6_6001 = ____D_72B6_6001_8868[jass.GetHandleId(_____82F1_96C4)]
    if _____72B6_6001 == nil or _____72B6_6001["剩余次数"] <= 0 then
        return false
    end
    _____72B6_6001["剩余次数"] = _____72B6_6001["剩余次数"] - 1
    _____5237_65B0D_663E_793A(_____82F1_96C4, _____72B6_6001)
    return true
end
--- 消费 D 的全部剩余强化次数（R 终式用），返回实际消费次数
____exports["消费全部D强化"] = function(_____82F1_96C4)
    if _____82F1_96C4 == nil or _____82F1_96C4 == 0 then
        return 0
    end
    local _____72B6_6001 = ____D_72B6_6001_8868[jass.GetHandleId(_____82F1_96C4)]
    if _____72B6_6001 == nil then
        return 0
    end
    local _____6B21_6570 = _____72B6_6001["剩余次数"]
    if _____6B21_6570 <= 0 then
        return 0
    end
    _____72B6_6001["剩余次数"] = 0
    _____5237_65B0D_663E_793A(_____82F1_96C4, _____72B6_6001)
    return _____6B21_6570
end
--- R 收束 / 主动结束 D（移除 Buff、刀环、计时器与表项）
____exports["结束D秘传"] = function(_____82F1_96C4)
    _____79FB_9664D_72B6_6001(_____82F1_96C4)
end
--- 获取 D 剩余强化次数（R 判定用）
____exports["获取D剩余强化次数"] = function(_____82F1_96C4)
    if _____82F1_96C4 == nil or _____82F1_96C4 == 0 then
        return 0
    end
    local _____72B6_6001 = ____D_72B6_6001_8868[jass.GetHandleId(_____82F1_96C4)]
    return _____72B6_6001 ~= nil and _____72B6_6001["剩余次数"] or 0
end
local function _____7834_7EFD_65A9_5EF6_957FD(_____7EA2_53F6, ______76EE_6807)
    if _____7EA2_53F6 == nil or _____7EA2_53F6 == 0 then
        return
    end
    local _____72B6_6001 = ____D_72B6_6001_8868[jass.GetHandleId(_____7EA2_53F6)]
    if _____72B6_6001 == nil then
        return
    end
    if _____72B6_6001["延长次数"] >= ____D_914D_7F6E["最大延长次数"] then
        return
    end
    _____72B6_6001["延长次数"] = _____72B6_6001["延长次数"] + 1
    _____72B6_6001["到期时间"] = _____72B6_6001["到期时间"] + ____D_914D_7F6E["延长秒"]
    if _____72B6_6001["到期回调ID"] ~= 0 then
        removeDelayedCallback(_____72B6_6001["到期回调ID"])
    end
    _____72B6_6001["到期回调ID"] = addDelayedCallback(
        (_____72B6_6001["到期时间"] - getGameTime()) * 1000,
        function()
            _____79FB_9664D_72B6_6001(_____7EA2_53F6)
        end
    )
    _____5237_65B0D_663E_793A(_____7EA2_53F6, _____72B6_6001)
end
local _____5DF2_6CE8_518C = false
____exports["注册朱雀院红叶D"] = function()
    debugLogForce(
        "红叶-D",
        "注册",
        "名称",
        "D",
        "函数",
        "注册朱雀院红叶D"
    )
    if _____5DF2_6CE8_518C then
        return
    end
    _____5DF2_6CE8_518C = true
    _____6CE8_518C_7834_7EFD_65A9_76D1_542C(_____7834_7EFD_65A9_5EF6_957FD)
    _____6CE8_518C_5355_4F4D_6280_80FD_58F3_76D1_542C({
        ["名称"] = "朱雀院红叶-秘传三式（D）",
        ["单位类型ID"] = _____82F1_96C4_5355_4F4D_7C7B_578BID,
        ["技能ID"] = "AMD1",
        ["获取或创建上下文"] = function(unit)
            return {["英雄"] = unit}
        end,
        ["释放技能"] = _____5F00_542FD_79D8_4F20_4E09_5F0F,
        ["创建独立技能实例"] = false
    })
end
____exports["朱雀院红叶D模块"] = {["技能ID"] = _____6731_96C0_9662_7EA2_53F6_6280_80FD_914D_7F6E.D["技能ID"], ["持续秒"] = ____D_914D_7F6E["持续秒"], ["强化次数"] = ____D_914D_7F6E["强化次数"], ["注册"] = ____exports["注册朱雀院红叶D"]}
return ____exports

File: TS/系统/03．技能系统/05．单位技能/04．英雄技能/21．朱雀院红叶/07．D技能.ts
/** @noSelfInFile */

import {
  朱雀院红叶技能配置,
  朱雀院红叶表现配置,
  朱雀院红叶音效配置,
  朱雀院红叶Buff配置,
  朱雀院红叶动作配置,
  朱雀院红叶动作槽,
  朱雀院红叶待平衡数值,
} from "./00．配置";

const jass = require("jass.common") as any;
const { stringToFourCCSafe } = require("lib.扩展函数.封装函数.01．通用工具.01．FourCC转换安全版") as {
  stringToFourCCSafe: (this: void, id: string) => number;
};
const { getGameTime, addDelayedCallback, removeDelayedCallback } = require("系统.00．核心系统.05．中心计时器") as {
  getGameTime: (this: void) => number;
  addDelayedCallback: (this: void, delayMs: number, callback: (this: void) => void) => number;
  removeDelayedCallback: (this: void, id: number) => void;
};
const { 注册单位技能壳监听 } = require("系统.03．技能系统.00．技能模板+函数.04．机制组件.10．复杂战斗通用机制.16．单位技能壳监听注册器") as {
  注册单位技能壳监听: (this: void, 参数: any) => void;
};
const { registerManualBuff, 移除单位指定Buff } = require("系统.05．Buff系统.00．Buff系统") as {
  registerManualBuff: (this: void, target: any, buffID: string, durationSec: number, effectValue: number, extras?: any) => void;
  移除单位指定Buff: (this: void, target: any, buffID: string) => boolean;
};
const { 创建单位坐标跟随特效, 销毁单位坐标跟随特效 } = require("lib.扩展函数.封装函数.01．通用工具.03．特效") as {
  创建单位坐标跟随特效: (this: void, unit: any, modelPath: string, effectKey?: string, scale?: number, height?: number, animSpeed?: number, 动画索引?: number, 面向弧度?: number, RGB?: { 红: number; 绿: number; 蓝: number; 透明度?: number }) => any;
  销毁单位坐标跟随特效: (this: void, unit: any, effectKey?: string) => void;
};
const { Sound3DII_UnitPlayReuse } = require("lib.扩展函数.封装函数.02．音效系统.03．3D音效播放") as {
  Sound3DII_UnitPlayReuse: (this: void, path: string, unit: any, cutoff: number) => any;
};
const { 播放英雄技能喊话 } = require("系统.09．表现系统.10．英雄语音.10．技能喊话.01．英雄技能喊话") as {
  播放英雄技能喊话: (this: void, 施法者: any, 英雄名: string, 技能ID: string) => boolean;
};
const {
  是朱雀院红叶,
  登记朱雀院清理,
  注册破绽斩监听,
  播放红叶动作,
} = require("./02．被动效果") as {
  是朱雀院红叶: (this: void, unit: any) => boolean;
  登记朱雀院清理: (this: void, 英雄: any, 名称: string, 清理: () => void) => void;
  注册破绽斩监听: (this: void, 回调: (this: void, 红叶: any, 目标: any) => void) => void;
  播放红叶动作: (this: void, 英雄: any, 槽: { 索引: number; 持续秒: number }) => void;
};
const { debugLogForce } = require("lib.扩展函数.自定义扩展函数.03．调试输出") as {
  debugLogForce: (this: void, module: string, ...args: any[]) => void;
};

const 英雄单位类型ID = stringToFourCCSafe(朱雀院红叶技能配置.单位类型ID);
const 秘传BuffID = 朱雀院红叶Buff配置.秘传三式;
const D配置 = 朱雀院红叶待平衡数值.D;
const D秘传三式音效 = 朱雀院红叶音效配置.D秘传三式;
const 刀环特效键 = "朱雀院红叶D刀环";

//=============================================================================
// D 秘传三式：8 秒、3 次强化资源
//=============================================================================

interface D状态 {
  剩余次数: number;
  到期时间: number;
  延长次数: number;
  到期回调ID: number;
}

const D状态表: Record<number, D状态 | undefined> = {};

function 刷新D显示(this: void, 英雄: any, 状态: D状态): void {
  const 剩余秒 = 状态.到期时间 - getGameTime();
  if (剩余秒 <= 0) {
    移除D状态(英雄);
    return;
  }
  registerManualBuff(英雄, 秘传BuffID, 剩余秒, 状态.剩余次数, { stack: 状态.剩余次数 });
}

function 移除D状态(this: void, 英雄: any): void {
  if (英雄 == null || 英雄 === 0) return;
  const id = jass.GetHandleId(英雄);
  const 状态 = D状态表[id];
  if (状态 == null) return;
  if (状态.到期回调ID !== 0) {
    removeDelayedCallback(状态.到期回调ID);
    状态.到期回调ID = 0;
  }
  debugLogForce("红叶-D", "Buff", "操作", "移除", "目标", 英雄);
  销毁单位坐标跟随特效(英雄, 刀环特效键);
  移除单位指定Buff(英雄, 秘传BuffID);
  delete D状态表[id];
}

function 开启D秘传三式(this: void, _context: any, 施法者: any, _技能实例ID: number | undefined): void {
  debugLogForce("红叶-D", "释放", "技能实例ID", "-");
  if (!是朱雀院红叶(施法者)) return;
  播放红叶动作(施法者, 朱雀院红叶动作槽.D启动);
  // 技能喊话：施法成功起点（全局 3D；随机二选一由喊话系统驱动；重复 D 刷新同样视为成功施法）
  播放英雄技能喊话(施法者, "朱雀院红叶", 朱雀院红叶技能配置.D.技能ID);
  const id = jass.GetHandleId(施法者);
  const 已有 = D状态表[id];
  if (已有 != null) {
    // 重复 D：刷新持续时间，次数保持（不叠加到 6 次）
    if (已有.到期回调ID !== 0) removeDelayedCallback(已有.到期回调ID);
    已有.到期时间 = getGameTime() + D配置.持续秒;
    已有.到期回调ID = addDelayedCallback(D配置.持续秒 * 1000, function D到期(this: void): void {
      移除D状态(施法者);
    });
    刷新D显示(施法者, 已有);
    return;
  }
  const 状态: D状态 = {
    剩余次数: D配置.强化次数,
    到期时间: getGameTime() + D配置.持续秒,
    延长次数: 0,
    到期回调ID: 0,
  };
  状态.到期回调ID = addDelayedCallback(D配置.持续秒 * 1000, function D到期(this: void): void {
    移除D状态(施法者);
  });
  D状态表[id] = 状态;
  debugLogForce("红叶-D", "状态", "开启秘传", 状态.剩余次数);
  刷新D显示(施法者, 状态);
  // 秘传刀环表现（模型/缩放/高度/RGB 全由表现配置驱动；秘传结束 移除D状态 统一销毁）
  if ((朱雀院红叶表现配置.D刀环.模型路径 as string) !== "") {
    创建单位坐标跟随特效(
      施法者,
      朱雀院红叶表现配置.D刀环.模型路径,
      刀环特效键,
      朱雀院红叶表现配置.D刀环.缩放,
      朱雀院红叶表现配置.D刀环.高度,
      1,
      undefined,
      0,
      朱雀院红叶表现配置.D刀环.RGB,
    );
  }
  // 秘传三式启动音（单次激活成功时一次；单位绑定，参数配置驱动；重复 D 刷新不重播，强化消费不单独响）
  Sound3DII_UnitPlayReuse(D秘传三式音效.路径, 施法者, D秘传三式音效.裁断距离);
  登记朱雀院清理(施法者, "红叶D", function D清理(this: void): void {
    移除D状态(施法者);
  });
}

//=============================================================================
// 强化消费接口（Q/W/E/R 在真正进入强化分支时调用）
//=============================================================================

/** 尝试消费 1 次 D 强化（无 D 状态或次数不足返回 false，技能仍执行基础效果） */
export function 尝试消费D强化(this: void, 英雄: any): boolean {
  if (英雄 == null || 英雄 === 0) return false;
  const 状态 = D状态表[jass.GetHandleId(英雄)];
  if (状态 == null || 状态.剩余次数 <= 0) return false;
  状态.剩余次数 = 状态.剩余次数 - 1;
  刷新D显示(英雄, 状态);
  return true;
}

/** 消费 D 的全部剩余强化次数（R 终式用），返回实际消费次数 */
export function 消费全部D强化(this: void, 英雄: any): number {
  if (英雄 == null || 英雄 === 0) return 0;
  const 状态 = D状态表[jass.GetHandleId(英雄)];
  if (状态 == null) return 0;
  const 次数 = 状态.剩余次数;
  if (次数 <= 0) return 0;
  状态.剩余次数 = 0;
  刷新D显示(英雄, 状态);
  return 次数;
}

/** R 收束 / 主动结束 D（移除 Buff、刀环、计时器与表项） */
export function 结束D秘传(this: void, 英雄: any): void {
  移除D状态(英雄);
}

/** 获取 D 剩余强化次数（R 判定用） */
export function 获取D剩余强化次数(this: void, 英雄: any): number {
  if (英雄 == null || 英雄 === 0) return 0;
  const 状态 = D状态表[jass.GetHandleId(英雄)];
  return 状态 != null ? 状态.剩余次数 : 0;
}

//=============================================================================
// 破绽斩延长（最多两次）
//=============================================================================

function 破绽斩延长D(this: void, 红叶: any, _目标: any): void {
  if (红叶 == null || 红叶 === 0) return;
  const 状态 = D状态表[jass.GetHandleId(红叶)];
  if (状态 == null) return;
  if (状态.延长次数 >= D配置.最大延长次数) return;
  状态.延长次数 = 状态.延长次数 + 1;
  状态.到期时间 = 状态.到期时间 + D配置.延长秒;
  if (状态.到期回调ID !== 0) removeDelayedCallback(状态.到期回调ID);
  状态.到期回调ID = addDelayedCallback((状态.到期时间 - getGameTime()) * 1000, function D延长到期(this: void): void {
    移除D状态(红叶);
  });
  刷新D显示(红叶, 状态);
}

//=============================================================================
// 注册入口（幂等）
//=============================================================================

let 已注册 = false;

export function 注册朱雀院红叶D(this: void): void {
  debugLogForce("红叶-D", "注册", "名称", "D", "函数", "注册朱雀院红叶D");
  if (已注册) return;
  已注册 = true;
  注册破绽斩监听(破绽斩延长D);
  注册单位技能壳监听({
    名称: "朱雀院红叶-秘传三式（D）",
    单位类型ID: 英雄单位类型ID,
    技能ID: "AMD1",
    获取或创建上下文: function D上下文(this: void, unit: any): any {
      return { 英雄: unit };
    },
    释放技能: 开启D秘传三式,
    创建独立技能实例: false,
  });
}

export const 朱雀院红叶D模块 = {
  技能ID: 朱雀院红叶技能配置.D.技能ID,
  持续秒: D配置.持续秒,
  强化次数: D配置.强化次数,
  注册: 注册朱雀院红叶D,
} as const;
